lib/sync/installation.js

```diff
@@ -34,8 +35,11 @@ module.exports.processInstallation = (app, queues) => {
     if (edges.length > 0) {
       // there's more data to get
       subscription.set(`repoSyncState.repos.${repositoryId}.${getCursorKey(task)}`, edges[edges.length - 1].cursor)
+      await subscription.save()
+
       const { removeOnComplete, removeOnFail } = job.opts
-      queues.installation.add(job.data, { removeOnComplete, removeOnFail })
+      const delay = Number(process.env.LIMITER_PER_INSTALLATION) || 1000
```

Comment:
Thought we'd reuse the `LIMITER_PER_INSTALLATION` here since it helps us not hit abuse limits as quickly. Basically all jobs will be added to the queue with this default delay, and we get the added benefit of it continuing to work the same if we add more workers. @gimenete any thoughts on this approach?

Reply:
I don't think hitting the rate limiting quickly is bad per se. If a sync requires 6000 API requests, I prefer to do 5000 API requests as quickly as possible, hit the rate limit, wait for the next reset and make 1000 API requests as fast as possible.

By delaying always jobs we guarantee that we don't hit the rate limit but we end up doing a lot less processing that we could do, I think.

Adding new workers should not be a problem since we now only have one job per installation. So either that job is being processed or not, but it's one job maximum per installation at the same time.

Reply:
The problem I'm trying to avoid with this is abuse rate limits, which have no direct correlation to the 5000 API requests per hour. In my testing against https://github.com/fusion-qa/, it was consistently triggering abuse detection within 2-3 minutes of the start of the job, and then we have to back off for at least 60 seconds when that is triggered. This allows us to smooth that out and not wait ~60 seconds every few minutes because we're hitting abuse limits. I've yet to ever see us come near the 5000/hr limit, but it's much easier to hit the abuse limit with this app due to the cost of he GraphQL queries.

Reply:
Sounds good 👍